contracts/SnapshotToken/Extensions/SimpleVote.sol

```diff
@@ -0,0 +1,191 @@
+    // % of tokens to participate in a voting to make it valid
+    uint256 public QUORUM = 40;
+    uint256 public VOTING_PERIOD;
+
+    /////////////////////////
+    // Mutable state
+    ////////////////////////
+    mapping (uint256 => Proposal) private _proposals;
+    mapping (address => bool) private _hasActiveProposal;
+    uint256 private _nextProposalId;
+
+    /////////////////////////
+    // Events
+    ////////////////////////
+    event LogNewProposal(address indexed owner, uint256 indexed proposalId, uint256 snapshotId, uint256 endTime);
+    event LogProposalResult(uint256 indexed proposalId, bool hasPassed, uint256 inFavor, uint256 against);
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    /// @param token which balances are used for voting
+    /// @param votingPeriodInDays amount of days that each vote will take
+    constructor(
+        ISnapshotableToken token,
+        uint256 votingPeriodInDays
+    )
+        public
+    {
+        VOTING_PERIOD = votingPeriodInDays * 1 days;
+        TOKEN = token;
+    }
+
+    /////////////////////////
+    // Public functions
+    ////////////////////////
+
+    /// @dev Creates a proposal, uniquely identifiable by its assigned proposalId
+    function addProposal() public {
+        require(!_hasActiveProposal[msg.sender], "Only one active proposal per address");
+
+        uint256 sId = TOKEN.currentSnapshotId();
+        // advance snapShotID, so that any subsequent trades will be in the non-finalized snapshot
+        TOKEN.createSnapshot();
+
+        _proposals[_nextProposalId++] = Proposal({
+            inFavor: 0,
+            against:0,
+            endTime: now + VOTING_PERIOD,
+            owner: msg.sender,
+            snapshotId: sId
+        });
+
+        _hasActiveProposal[msg.sender] = true;
+
+        emit LogNewProposal(msg.sender, _nextProposalId - 1, sId, now + VOTING_PERIOD);
+    }
+
+    /// @dev increase the votecount on a given proposal by the token balance of the sender
+    ///   throws if proposal does not exist or the vote on it has ended already. Votes are final,
+    ///   changing the vote is not allowed
+    /// @param proposalId of the proposal to be voted on
+    /// @param voteInFavor of the desired proposal
+    function vote(uint256 proposalId, bool voteInFavor) public {
+        incrementVote(proposalId, voteInFavor, msg.sender);
+    }
+
+    /// @dev same as vote, only for a relayed vote. Will throw if provided signature (v,r,s) does not match
+    ///  the address of the voter
+    /// @param voter address whose token balance should be used as voting weight
+    function relayedVote(
+        uint256 proposalId,
+        bool voteInFavor,
+        address voter,
+        bytes32 r,
+        bytes32 s,
+        uint8 v
+    )
+        public
+    {
+        // check that message signature matches the voter address
+        // solium-disable indentation
+        require(ecrecover(
```

Comment:
👍 for following "old" EIP but not getting into https://github.com/alexvandesande/EIPs/blob/ee2347027e94b93708939f2e448447d030ca2d76/EIPS/eip-1077.md and sign typed data as for example Gnosis Safe is doing. 
the objective here was to minimize gas usage to make voting affordable and and some lean format is the way to do it

Reply:
ok if you provide `voter` in parameter you should check if the erecover gives you same address

now actually relayer can use valid message to vote on someone else as `voter` is not a part of the message